rules/Symfony73/Rector/Class_/GetFiltersToAsTwigFilterAttributeRector.php

```diff
@@ -0,0 +1,210 @@
+ */
+final class GetFiltersToAsTwigFilterAttributeRector extends AbstractRector
+{
+    public function __construct(
+        private readonly ReflectionProvider $reflectionProvider
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Changes getFilters() in TwigExtension to #[TwigFilter] marker attribute above function',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+use Twig\Extension\AbstractExtension;
+
+class SomeClass extends AbstractExtension
+{
+    public function getFilters()
+    {
+        return [
+            new \Twig\TwigFilter('filter_name', [$this, 'localMethod']),
+        ];
+    }
+
+    public function localMethod($value)
+    {
+        return $value;
+    }
+}
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFilter;
+
+class SomeClass extends AbstractExtension
+{
+    #[TwigFilter('filter_name')]
+    public function localMethod($value)
+    {
+        return $value;
+    }
+}
+CODE_SAMPLE
+                )]
+        );
+    }
+
+    public function getNodeTypes(): array
+    {
+        return [Class_::class];
+    }
+
+    /**
+     * @param Class_ $node
+     */
+    public function refactor(Node $node): ?Class_
+    {
+        if (! $this->reflectionProvider->hasClass(TwigClass::TWIG_EXTENSION)) {
+            return null;
+        }
+
+        if ($node->isAbstract() || $node->isAnonymous()) {
+            return null;
+        }
+
+        if (! $this->isObjectType($node, new ObjectType(TwigClass::TWIG_EXTENSION))) {
+            return null;
+        }
+
+        $getFilterMethod = $node->getMethod('getFilters');
+        if (! $getFilterMethod instanceof ClassMethod) {
+            return null;
+        }
+
+        $hasChanged = false;
+
+        foreach ((array) $getFilterMethod->stmts as $stmt) {
+            // handle return array simple case
+            if (! $stmt instanceof Return_) {
+                continue;
+            }
+
+            if (! $stmt->expr instanceof Array_) {
+                continue;
+            }
+
+            $returnArray = $stmt->expr;
+            foreach ($returnArray->items as $key => $arrayItem) {
+                if (! $arrayItem->value instanceof New_) {
+                    continue;
+                }
+
+                $new = $arrayItem->value;
+                if (! $this->isObjectType($new->class, new ObjectType(TwigClass::TWIG_FILTER))) {
+                    continue;
+                }
+
+                if (count($new->getArgs()) !== 2) {
+                    continue;
+                }
+
+                $secondArg = $new->getArgs()[1];
+                if ($secondArg->value instanceof MethodCall && $secondArg->value->isFirstClassCallable()) {
+                    throw new ShouldNotHappenException('Not supported yet');
+                }
```

Comment:
the syntax seems should be:

```php
new \Twig\TwigFilter('some_filter', $this->localMethod(...)),
```

I will look into it

Reply:
I created PR:

- https://github.com/rectorphp/rector-symfony/pull/763

for it.